provider/shim/shim.go

```diff
@@ -2,14 +2,26 @@ package shim
 
 import (
 	"context"
+
+	pfprovider "github.com/hashicorp/terraform-plugin-framework/provider"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 	"github.com/hashicorp/terraform-provider-aws/internal/provider"
+	"github.com/hashicorp/terraform-provider-aws/internal/provider/fwprovider"
 )
 
-func NewProvider() *schema.Provider {
-	prov, err := provider.New(context.Background())
+type UpstreamProvider struct {
+	SDKV2Provider           *schema.Provider
+	PluginFrameworkProvider pfprovider.Provider
+}
+
+func NewUpstreamProvider(ctx context.Context) (UpstreamProvider, error) {
+	primary, err := provider.New(ctx)
 	if err != nil {
-		panic(err)
+		return UpstreamProvider{}, err
 	}
-	return prov
+	pf := fwprovider.New(primary)
```

Comment:
@iwahbe this is the state sharing that removes the config-related panic. This seems to work. We end up calling Configure twice through muxer code but it seems "idempotent enough".